### PR TITLE
Part8

### PR DESCRIPTION
**Pull request by dpsproservices**
Ready for pull request review.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -227,163 +147,92 @@ struct Numeric
...
-    // #11
-    template<typename DivisorType> 
-    Numeric& operator/=(const DivisorType& t)
+    template<typename OtherType> 
+    Numeric& operator/=(const OtherType& o) // #3
     {
-        // struct Numeric template type is int
-        if constexpr (std::is_same<Type,int>::value)
+        // template type is int
+        if constexpr (std::is_same<OtherType,int>::value)
```
Comment (matkatmusic) [inline main.cpp]: `template type` is referring to the class template type.
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -227,163 +147,92 @@ struct Numeric
...
         {
-            // function parameter type is also int
-            if constexpr (std::is_same<DivisorType,int>::value)
+            // function parameter type is int
+            if constexpr (std::is_same<decltype(o),int>::value)
```
Comment (matkatmusic) [inline main.cpp]: you already know the parameter's type: `const OtherType`. 
no need for `decltype` here
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -227,163 +147,92 @@ struct Numeric
...
-            else if ( t < std::numeric_limits<DivisorType>::epsilon() )
+            else if ( o < std::numeric_limits<OtherType>::epsilon() )
             {
                 // else if function parameter is less than epsilon dont do the divison
                 std::cerr << "can't divide integers by zero!" << std::endl;
                 return *this;
             }
         } 
-        else if ( t < std::numeric_limits<Type>::epsilon() )
+        else if ( o < std::numeric_limits<OtherType>::epsilon() )
```
Comment (matkatmusic) [inline main.cpp]: template type, not function parameter type.
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -227,163 +147,92 @@ struct Numeric
...
-    }
+    std::unique_ptr<Type> value;
 };
 
 template<typename Type> 
-void myNumericFreeFunct( std::unique_ptr<Type>& value)
+void cube( std::unique_ptr<Type>& value) // #7
 {
-    *value += 7;
+    *value = *value * *value * *value;
```
Comment (matkatmusic) [inline main.cpp]: use a local reference to make this look nicer:
```
auto& v = *value;
v = v * v * v;
```
- [ ] revise

File: main.cpp [hunk 5]
```diff
@@ -227,163 +147,92 @@ struct Numeric
...
+        if constexpr (std::is_same<OtherType,int>::value)
         {
-            // function parameter type is also int
-            if constexpr (std::is_same<DivisorType,int>::value)
+            // function parameter type is int
+            if constexpr (std::is_same<decltype(o),int>::value)
             {
                 // function parameter is integer 0 don't do the division
-                if (t == 0)
+                if (o == 0)
```
Comment (matkatmusic) [inline main.cpp]: Please use a different variable name besides `o`.  always try to make your code as self-documenting as possible.
I recommend: `otherType`
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 3 above)
Comment (dpsproservices) [reply main.cpp]: Done.

File: main.cpp (hunk 2 above)
Comment (dpsproservices) [reply main.cpp]: Done.

File: main.cpp (hunk 1 above)
Comment (dpsproservices) [reply main.cpp]: Done.

File: main.cpp (hunk 4 above)
Comment (dpsproservices) [reply main.cpp]: Done.

File: main.cpp (hunk 5 above)
Comment (dpsproservices) [reply main.cpp]: Done.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**